Transposed xcalloc arguments

### Problem

When compiling from master with `-Wcalloc-transposed-arguments` on a clean checkout, two lines trigger that warning:
```
-- Using NVIM_VERSION: v0.10.0-dev-3105+g17d8f3b32
[228/415] Building C object src/nvim/CMakeFiles/nvim_bin.dir/decoration.c.o
/home/cnihelton/Dev/2_Tools/neovim/src/nvim/decoration.c: In function ‘buf_signcols_count_range’:
/home/cnihelton/Dev/2_Tools/neovim/src/nvim/decoration.c:793:31: warning: ‘xcalloc’ sizes specified with ‘sizeof’ in the earlier argument and not in the later argument [-Wcalloc-transposed-args]
  793 |   int *count = xcalloc(sizeof(int), (size_t)(row2 + 1 - row1));
      |                               ^~~
/home/cnihelton/Dev/2_Tools/neovim/src/nvim/decoration.c:793:31: note: earlier argument should specify number of elements, later size of each element
[344/415] Building C object src/nvim/CMakeFiles/nvim_bin.dir/strings.c.o
/home/cnihelton/Dev/2_Tools/neovim/src/nvim/strings.c: In function ‘adjust_types’:
/home/cnihelton/Dev/2_Tools/neovim/src/nvim/strings.c:946:47: warning: ‘xcalloc’ sizes specified with ‘sizeof’ in the earlier argument and not in the later argument [-Wcalloc-transposed-args]
  946 |                              ? xcalloc(sizeof(const char *), (size_t)arg)
      |                                               ^~~~~
/home/cnihelton/Dev/2_Tools/neovim/src/nvim/strings.c:946:47: note: earlier argument should specify number of elements, later size of each element
```

Looking at the code the arguments seem transposed indeed, as we have `sizeof` a type as the first argument.

![image](https://github.com/neovim/neovim/assets/11138291/80fa9f94-6dfb-42a9-be4f-be36af1ab19b)
![image](https://github.com/neovim/neovim/assets/11138291/6756286a-201a-44c5-8a71-f8b68e138c49)

While the transposing is harmless, no other calls to xcalloc are transposed like that, so I'd assume it's a code style issue that requires fixing.

### Steps to reproduce

git clone https://github.com/neovim/neovim
cd neovim
make CMAKE_BUILD_TYPE=RelWithDebInfo -j4

### Expected behavior

No warnings related to `calloc` family of functions.

### Neovim version (nvim -v)

NVIM_VERSION: v0.10.0-dev-3105+g17d8f3b32

### Vim (not Nvim) behaves the same?

no, not applicable

### Operating system/version

Ubuntu 24.04 LTS on WSL2

### Terminal name/version

Windows Terminal v1.20.11271.0

### $TERM environment variable

xterm-256color

### Installation

Building from source

--- a/src/nvim/decoration.c
+++ b/src/nvim/decoration.c
@@ -790,7 +790,7 @@ void buf_signcols_count_range(buf_T *buf, int row1, int row2, int add, TriState 
   }
 
   // Allocate an array of integers holding the number of signs in the range.
-  int *count = xcalloc(sizeof(int), (size_t)(row2 + 1 - row1));
+  int *count = xcalloc((size_t)(row2 + 1 - row1), sizeof(int));
   MarkTreeIter itr[1];
   MTPair pair = { 0 };
 

--- a/src/nvim/strings.c
+++ b/src/nvim/strings.c
@@ -943,7 +943,7 @@ static int adjust_types(const char ***ap_types, int arg, int *num_posarg, const 
 {
   if (*ap_types == NULL || *num_posarg < arg) {
     const char **new_types = *ap_types == NULL
-                             ? xcalloc(sizeof(const char *), (size_t)arg)
+                             ? xcalloc((size_t)arg, sizeof(const char *))
                              : xrealloc(*ap_types, (size_t)arg * sizeof(const char *));
 
     for (int idx = *num_posarg; idx < arg; idx++) {

